exchange/index.go

```diff
@@ -36,7 +47,7 @@ const KIndex = "idx"
 type Index struct {
 	ds     datastore.Batching
 	root   *hamt.Node
-	bstore blockstore.Blockstore
+	bstore blockstore.GCBlockstore
```

Comment:
This should stay a regular blockstore interface and the GCBlockstore should be accessed via type assertion. You actually want to wrap the global blockstore so anyone can claim the lock when needed otherwise the lock is useless. So in `options.go:Options::fillDefaults` you can first check if a blockstore is provided that it is already a GC: `gcbl, ok := opts.Blockstore.(blockstore.GCBlockstore)` if not ok then wrap it up.
``

Reply:
Also tx.Close() should grab the lock on the blockstore.

Reply:
I'm still confused about this actually, in the Index, we don't receive a blockstore from the exchange, so we create it

`idx.bstore = blockstore.NewBlockstore(idx.ds)` but it fails in the GC because it's not a GCBlockstore

should we do instead ?
`idx.bstore = blockstore.NewGCBlockstore(blockstore.NewBlockstore(ds), blockstore.NewGCLocker())`

or receive a gcblockstore in parameter of NewIndex ?